jgrapht-core/src/main/java/org/jgrapht/alg/spanning/PrimMinimumSpanningTree.java

```diff
@@ -31,7 +31,7 @@
  * @param <V> the graph vertex type
  * @param <E> the graph edge type
  *
- * @author Alexey Kudinkin
+ * @author Alexey Kudinkin, Alexandru Valeanu
  * @since Mar 5, 2013
  */
 public class PrimMinimumSpanningTree<V, E>
```

Comment:
Use a new class with an appropriate name for this algorithm (something with Dense inside). 

Add proper documentation to explain to the users that this is faster if their graph is dense and also link between the two separate Prim implementations by saying something like "if your graph is dense consider the ..." and "if your graph is sparse consider ..." .